src/pyupgrade/fixes.rs

```diff
@@ -132,3 +132,35 @@ pub fn remove_super_arguments(locator: &SourceCodeLocator, expr: &Expr) -> Optio
 
     None
 }
+
+/// U011 -fix
+pub fn remove_unnecessary_lur_cache_params(
+    locator: &SourceCodeLocator,
+    decor_at: Location,
+) -> Option<Fix> {
+    let contents = locator.slice_source_code_at(&decor_at);
+    let fix_start = lexer::make_tokenizer(&contents)
+        .flatten()
+        .find_map(|(start, tok, _)| {
+            if matches!(tok, Tok::Lpar) {
+                Some(helpers::to_absolute(&start, &decor_at))
+            } else {
+                None
+            }
+        });
+
+    let fix_end = lexer::make_tokenizer(&contents)
+        .flatten()
+        .find_map(|(_, tok, end)| {
+            if matches!(tok, Tok::Rpar) {
+                Some(helpers::to_absolute(&end, &decor_at))
+            } else {
+                None
+            }
+        });
```

Comment:
The code isn't as nice, but we could this in one pass with a loop, like:

```
let mut fix_start: Option<Location> = None;
let mut fix_end: Option<Location> = None;
for (start, tok, end) in lexer::make_tokenizer(&contents).flatten() {
  ...
}
```

Reply:
So, one case this doesn't handle, is this:

```py
@lru_cache(())
def fixme():
    pass
```

(I know that `checks::unnecessary_lru_cache_params` wouldn't actually trigger in this case, but it is something that this fix should be robust to.)

The basic issue is that we're looking for the first left-paren, and then the first right-paren. We need to do something like in the `// Case 1: `object` is the only base.` example at the top of this file: track the number of open parens, to ensure that we're ending as soon as we close that first paren.

Reply:
Sure. In fact, I started with this fix, but went ahead with more functional way of finding the first pair of brackets. I am glad to change it to suggested, it will be faster and more robust. 👍